pyccel/codegen/printing/cwrappercode.py

```diff
@@ -877,6 +889,9 @@ def _print_FunctionDef(self, expr):
 
         # Call free function for python type
         wrapper_body += [FunctionCall(Py_DECREF, [i]) for i in self._to_free_PyObject_list]
+        # Call free function for C type
+        if self._target_language == 'c':
+            wrapper_body += [Deallocate(i) for i in expr.arguments if i.rank > 0]
```

Comment:
Here (FunctionDef) you deallocate if `i.rank > 0` but above (Interface) you deallocate if `i.allocatable`. Is there any reason for this difference?

Reply:
no, there is no difference, the variable is allocatable if rank > 0, so I check directly the rank.

Reply:
Then I think you should do the same thing in both places